packages/@aws-cdk/aws-glue/lib/database.ts

```diff
@@ -0,0 +1,142 @@
+import s3 = require('@aws-cdk/aws-s3');
+import cdk = require('@aws-cdk/cdk');
+import { CfnDatabase } from './glue.generated';
+
+export interface IDatabase extends cdk.IConstruct {
+  /**
+   * The ARN of the catalog.
+   */
+  readonly catalogArn: string;
+
+  /**
+   * The catalog id of the database (usually, the AWS account id)
+   */
+  readonly catalogId: string;
+
+  /**
+   * The ARN of the database.
+   */
+  readonly databaseArn: string;
+
+  /**
+   * The name of the database.
+   */
+  readonly databaseName: string;
+
+  /**
+   * The location of the database (for example, an HDFS path).
+   */
+  readonly locationUri: string;
+
+  export(): DatabaseImportProps;
+}
+
+export interface DatabaseImportProps {
+  catalogArn: string;
+  catalogId: string;
+  databaseArn: string;
+  databaseName: string;
+  locationUri: string;
+}
+
+export interface DatabaseProps {
+  /**
+   * The name of the database.
+   */
+  databaseName: string;
+
+  /**
+   * The location of the database (for example, an HDFS path).
+   *
+   * @default a bucket is created and the database is stored under s3://<bucket-name>/<database-name>
+   */
+  locationUri?: string;
+}
+
+/**
+ * A Glue database.
+ */
+export class Database extends cdk.Construct {
+  /**
+   * Creates a Database construct that represents an external database.
+   *
+   * @param scope The scope creating construct (usually `this`).
+   * @param id The construct's id.
+   * @param props A `DatabaseImportProps` object. Can be obtained from a call to `database.export()` or manually created.
+   */
+  public static import(scope: cdk.Construct, id: string, props: DatabaseImportProps): IDatabase {
+    return new ImportedDatabase(scope, id, props);
+  }
+
+  public readonly catalogArn: string;
+  public readonly catalogId: string;
+  public readonly databaseArn: string;
+  public readonly databaseName: string;
+  public readonly locationUri: string;
+
+  constructor(scope: cdk.Construct, id: string, props: DatabaseProps) {
+    super(scope, id);
+
+    if (props.locationUri) {
+      this.locationUri = props.locationUri;
+    } else {
+      const bucket = new s3.Bucket(this, 'Bucket');
```

Comment:
Does it make sense to use the same default bucket for all tables in the database (unless a bucket is specified)?

Reply:
It might, but I also think there's value is isolating databases's from each other - isolate encryption and policies etc., and isolate performance since high-volume tables can push S3's `listObjects` API to its limits. The same logic explains why a new `Bucket` is created (by default) for a new `Table`.

I'm honestly not 100% sure what the database's location is used for. I think that tables created by crawlers or through the UI might default their location to the database's location, but I'm not sure.

Reply:
okay, let's leave it like this. It's probably not that important